libraries/standard/http/src/private/http_client_internal.h

```diff
@@ -28,4 +28,84 @@
     #define IotLogDebugWithArgs( format, ... )
 #endif /* ifdef USE_AWS_IOT_CSDK_LOGGING */
 
+/**
+ * @brief The HTTP protocol version of this library is HTTP/1.1.
+ */
+#define HTTP_PROTOCOL_VERSION              "HTTP/1.1"
+
+/**
+ * @brief Default value when pRequestInfo->pPath == NULL.
+ */
+#define HTTP_EMPTY_PATH                    "/"
+#define HTTP_EMPTY_PATH_LEN                ( uint8_t ) ( sizeof( HTTP_EMPTY_PATH ) - 1 )
+
+/**
+ * @brief Consants for HTTP header formatting
+ */
+#define HTTP_HEADER_LINE_SEPARATOR         "\r\n"
+#define CARRIAGE_RETURN_CHARACTER          '\r'
+#define NEWLINE_CHARACTER                  '\n'
+#define HTTP_HEADER_FIELD_SEPARATOR        ": "
+#define COLON_CHARACTER                    ':'
+#define SPACE_CHARACTER                    ' '
+#define EQUAL_CHARACTER                    '='
+#define DASH_CHARACTER                     '-'
+#define HTTP_HEADER_LINE_SEPARATOR_LEN     ( uint8_t ) ( sizeof( HTTP_HEADER_LINE_SEPARATOR ) - 1 )
+#define HTTP_HEADER_FIELD_SEPARATOR_LEN    ( uint8_t ) ( sizeof( HTTP_HEADER_FIELD_SEPARATOR ) - 1 )
+#define COLON_CHARACTER_LEN                ( uint8_t ) ( sizeof( COLON_CHARACTER ) - 1 )
+#define SPACE_CHARACTER_LEN                ( uint8_t ) ( sizeof( SPACE_CHARACTER ) - 1 )
+#define EQUAL_CHARACTER_LEN                ( uint8_t ) ( sizeof( EQUAL_CHARACTER ) - 1 )
+#define DASH_CHARACTER_LEN                 ( uint8_t ) ( sizeof( DASH_CHARACTER ) - 1 )
+
+/**
+ * @brief The maximum length for a 32-bit integer when converted to a string.
+ *
+ * This is used to initialize a local array for the final headers to send.
+ */
+#define INT32_STRING_MAX_LEN               ( uint8_t ) ( sizeof( INT32_STRING_MAX ) - 1 )
```

Comment:
`sizeof(INT32_STRING_MAX)` is the size of a 32 bit value i.e. 4 bytes.  This value will resolve to 3. 
I think you want to calculate the number of digits in the max value, whose value is **10**.
You cannot calculate it at compile-time, you will have to just store the value directly.

Reply:
Oh my bad. This is what happens when you use regex search and replace lol. You are right, this was originally set to the 10.